[Exercise-2] Otp-Field/src/components/OtpInputs.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import {StyleSheet, View, TextInput} from 'react-native';
+import {Global, Colors, Spacing} from '../styles';
+
+const numRegex = /[0-9]/;
+
+const createRef = (inputs) => inputs.map(() => ({ref: React.createRef()}));
+
+class OtpInputs extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      otp: createRef(props.inputs),
+    };
+  }
+
+  focusInput(index, value) {
+    const {otp} = this.state;
+    const {getOtp} = this.props;
+
+    if (value == '' && index != 0) {
+      otp[index - 1].ref.current.focus();
+    }
+
+    const isNumericValue = numRegex.test(value);
+    if (!isNumericValue) {
+      return;
+    }
+    if (index < otp.length - 1) {
+      otp[index + 1].ref.current.focus();
+    }
+    otp[index]['value'] = value;
+    this.setState({otp});
+
+    if (index === otp.length - 1) {
+      otp[index].ref.current.blur();
+      const otpValue = otp.map(({value}) => value);
+      getOtp(otpValue.join(''));
+    }
+  }
+
+  activateInput(index) {
+    const {otp} = this.state;
```

Comment:
don't make direct changes to the state, instead create a new hash here using JSON deconstruction,